### PR TITLE
Remove inline Javascript for  administrator/components/com_plugins/tmpl/plugin/edit.php. 

### PR DESCRIPTION
**Pull request by astridx**
Pull Request for Issue # https://github.com/joomla-projects/joomla-es6/issues/55.

Is it correct that we first edit inline javascript and then continue with es6?

Please have a look at my first attempt to eliminate inline javascript.

Concerning es6: How shall we handle <tt>Unexpected use of 'self'</tt> and <tt>Unexpected use of 'top'</tt>

### CODE REVIEW

File: media/com_plugins/js/edit.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ * @copyright   Copyright (C) 2005 - 2018 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+Joomla.submitbutton = (task) => {
+  if (task === 'plugin.cancel' || document.formvalidator.isValid(document.getElementById('style-form'))) {
+    Joomla.submitform(task, document.getElementById('style-form'));
+  }
+
+  if (task !== 'plugin.apply') {
+    if (self !== top) {
```
Comment (dneukirchen) [inline media/com_plugins/js/edit.js]: Do you know where this is used?

Comment (dneukirchen) [reply media/com_plugins/js/edit.js]: or anybody else? @dgt41 ?

Comment (astridx) [reply media/com_plugins/js/edit.js]: @dneukirchen To be honest, I do not know. That is why I asked. I did not know where to test it.

As well as the script in administrator/components/com_plugins/tmpl/plugin/modal.php with which I wanted to start.

Comment (dneukirchen) [reply media/com_plugins/js/edit.js]: Well... if nobody (ping @dgt41 @C-Lodder @Fedik ) knows where this is actually used, we should remove it completely.

**Comment by dneukirchen** [comment]
> Is it correct that we first edit inline javascript and then continue with es6?

yes. 

> Concerning es6: How shall we handle Unexpected use of 'self' and Unexpected use of 'top'

use window.self and window.top

File: media/com_plugins/js/edit.js (hunk 1 above)
Comment (Fedik) [reply media/com_plugins/js/edit.js]: no idea, need to search,
I guess some modal/popup with an iframe for quick plugin editing, somewhere, 

upd. maybe in Finder or Redirects

Comment (dneukirchen) [reply media/com_plugins/js/edit.js]: I cant find any usage. So i guess wie can remove this inline script completely.

**Comment by astridx** [comment]
This code can only be used in a modal, right? I just searched all plugins. I did not find any occurrence. I think that's been built in for Custom Fields. For example, checkbox once used a modal. In the XML (/plugins/fields/checkboxes/checkboxes.xml) you sees this also at the class name (list_templates_modal). That is confusing. Should I change the classnames in the XML of custom fields, too? Here I will now delete the not used code.

If a third party plugin need this feature, it had to implement it by hisself.

I also did es6, so we do not have to touche it once more.

File: administrator/components/com_plugins/tmpl/plugin/edit.php [hunk 2]
```diff
@@ -22,22 +22,6 @@
 $isModal  = $input->get('layout') === 'modal' ? true : false;
```
Comment (astridx) [inline administrator/components/com_plugins/tmpl/plugin/edit.php:22]: I think that can also get out. But I'm not brave enough. What do the others think?

**Comment by dneukirchen** [comment]
unfortunately i found the usage: https://github.com/joomla-projects/joomla-es6/blob/dev/administrator/components/com_redirect/tmpl/links/default.php#L25

does not work in j4, but @Fedik was right :-(

**Comment by astridx** [comment]
@dneukirchen Fortunately, you found it. It would be worse if I deleted it and we would find the error later.

<s>So now I want to get the modal https://github.com/joomla-projects/joomla-es6/blob/dev/administrator/components/com_redirect/tmpl/links/default.php#L25
to run, so that I can test changes in this PR. But I am not sure how to do it. Can I check this out here: https://github.com/joomla-projects/joomla-es6/blob/9eb0472a4cffffe7b50d164a9cda1ff5ac99f9c1/layouts/joomla/form/field/media.php#L105</s>

**Comment by astridx** [comment]
I close this PR because I messed up here. I will open a new one later.

**Comment by dneukirchen** [comment]
If this is about the media plugin files, that is my fault and i already opened a PR for this #73 but waiting for @laoneo to have a look.

**Comment by astridx** [comment]
Yes, it was because of the files resize, crop and rotate which are in now in this PR but I did not change them in this PR.

**Pull request closed without merging**